### ISSUE TITLE
Do not reorder includes

### ISSUE THREAD

**Issue by Flamefire** [include-what-you-use/include-what-you-use#269]
The fix_includes.py script does to much work for my usecase. I just want it to remove unnecessary includes and add required ones or forward declarations. The sorting completely messes up the ordering and on the (huge) project the changes are to many to check. Most problematic is, that it moves headers around, that should stay on fixed positions.

So what I like: Do the suggested removes. Add new headers as suggested (and those, and only those sorted) to the end of the include list (or where it seems to be appropriate) but do not shuffle existing includes around. This way one can easily check, what was done via version control (add/delete are easy, moves are hard) and order them manually if required with iwyu not touching them afterwards

**Comment by phb** [comment]
The moving around of headers can also cause compilation issues. 
iwyu currently moves headers in to
1) extern "C" {
#include ...
}
blocks which obviously causes problems.
2) Headers that depend on #define's being set can end up being moved prior to where the #define is set.

**Comment by vsapsai** [comment]
I suggest to separate issues where reordering breaks the build from convenience of use with version control. I consider correctness issues to be of higher priority.

**Comment by Flamefire** [comment]
This is hard to do. For example I have a "prefix" header which must be the first include and a "suffix" header which must be the last one. This also varies by subprojects. So IMO easiest fix is to disable sorting via a switch. As a second step implement something that influences the sort order.

**Comment by kimgr** [comment]
I started thinking about this the other day, and I agree that sorting should be optional.

I even think it should be off by default, with a switch to enable it. IWYU causes enough mayhem as it is that we don't need to force Google's include order convention :-)

**Comment by vpoughon** [comment]
I would be very interested by this feature also :) Is there a workaround perhaps?

**Comment by Flamefire** [comment]
I remember simply removing the sorting manually. I guess it was the sort call here: https://github.com/include-what-you-use/include-what-you-use/blob/4a0cb6a5262243aad338b15e75dc713a1625b245/fix_includes.py#L1966

**Comment by Phantal** [comment]
Getting rid of `decorated_move_spans.sort()` doesn't have the intended effect.

The following appears to work:

```
+ 1932:  seen = set()
- 1933:  move_spans = set([fl.move_span for fl in file_lines if fl.move_span])
# this change is necessary to preserve order since sets are not ordered, however it does eliminate duplicates
+ 1933:  move_spans = [fl.move_span for fl in file_lines if fl.move_span and not (fl.move_span in seen or seen.add(fl.move_span))]

- 1966:  decorated_move_spans.sort()
+ 1966:  if flags.reorder:
+ 1967:    decorated_move_spans.sort()
+ 1968:  else:
# This makes it only pay attention to the line range and "kind" (system header, project header, forward decl, etc)
+ 1969:    decorated_move_spans.sort(key=lambda x: x[0:-2])

+ 2172:  parser.add_otion('--reorder', default=False, action='store_true', dest="reorder",
+ 2173:                    help=('Re-order lines'))
+ 2174:  parser.add_otion('--no-reorder', default=False, action='store_false', dest="reorder",
+ 2175:                    help=('Do not re-order lines'))
+ 2176:
```

... though I'll readily admit this is the first time I've ever written any python, so I wouldn't be surprised if there's a better way.

**Comment by kimgr** [comment]
Closing as duplicate of #35 (also possibly fixed by #569)
